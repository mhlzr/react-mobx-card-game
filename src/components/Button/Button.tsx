import React, { FunctionComponent, ReactElement } from 'react';
import styled from 'styled-components';

const ButtonLayout = styled.button`
    appearance: none;
    background-color: transparent;
    border: 1px solid currentColor;
    color: var(--secondary-color);
`;

interface ButtonProps {
    type?: string;
    onClick?(): void;
    children: JSX.Element[] | JSX.Element | string;
}

export const Button: FunctionComponent<ButtonProps> = ({ type, onClick, children }: ButtonProps): ReactElement => {
    return (
        <ButtonLayout type={type} onClick={onClick}>{children}</ButtonLayout>
    );
}
