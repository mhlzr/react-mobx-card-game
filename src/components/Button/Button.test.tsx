import React from 'react';
import { shallow } from 'enzyme'

import { Button } from './Button';

describe('Button', () => {
    it('renders with a label', () => {
        const wrapper = shallow(<Button>Test</Button>);
        expect(wrapper.text()).toEqual('Test');
    });

    it('renders with child-components', () => {
        const wrapper = shallow(<Button><span>Child</span>></Button>);
        expect(wrapper.find('span')).toHaveLength(1);
        expect(wrapper.find('span').text()).toEqual('Child');
    });

    it('has an onClick handler', () => {
        const onClick = jest.fn();
        const wrapper = shallow(<Button onClick={onClick}>Test</Button>);
        wrapper.simulate('click');
        expect(onClick).toHaveBeenCalled();
    });

    it('has a snapshot', () => {
        const wrapper = shallow(<Button>Test</Button>);
        expect(wrapper).toMatchSnapshot();
    });
})
