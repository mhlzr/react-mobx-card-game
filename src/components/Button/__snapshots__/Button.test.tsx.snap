// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button has a snapshot 1`] = `
<styled.button
  aria-pressed={false}
  disabled={false}
>
  Test
</styled.button>
`;
